File: ArchitectureComponentsMVVM/app/src/main/java/com/andr/mvvm/viewmodels/UserListViewModel.java
package com.andr.mvvm.viewmodels;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.andr.mvvm.repository.User;
import com.andr.mvvm.repository.UserRepository;

import java.util.List;

public class UserListViewModel extends AndroidViewModel {

    private UserRepository userRepository;
    private LiveData<List<User>> mAllUsers;

    public UserListViewModel(Application application){
        super(application);
        userRepository = new UserRepository(application);
        mAllUsers = userRepository.getAllUsers();
    }

    public LiveData<List<User>> getAllUsers() {
        return mAllUsers;
    }

    public void insert(User user) {
        userRepository.insert(user);
    }
}

File: MVVM_Kotlin/app/src/main/java/com/mvvm/kot/ViewModels/UserListViewModel.kt
package com.mvvm.kot.ViewModels

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import android.os.AsyncTask
import com.mvvm.kot.Repository.UserDBRepository
import com.mvvm.kot.Repository.UserDao

class UserListViewModel:AndroidViewModel {

    private  var userRepository: UserDBRepository
    private  var mAllUsers: LiveData<List<User>>


    constructor(application: Application) : super(application){
        userRepository = UserDBRepository(application)
        mAllUsers  = userRepository.getAllUsers()
    }

    fun getAllUsers(): LiveData<List<User>> {
        return mAllUsers
    }

    fun insert(user: User) {
        userRepository.insert(user)
    }



}